drivers/dma/dma_silabs_siwx917.c

```diff
@@ -90,9 +97,116 @@ static inline int siwx917_dma_addr_adjustment(uint32_t adjustment)
+			return -EINVAL;
+		}
+		/* Set the total number of DMA transfers */
+		cfg->totalNumOfDMATrans = block_addr->block_size - 1;
+		/* Set the transfer type based on whether it is a peripheral request */
+		cfg->transferType = siwx917_transfer_direction(config->channel_direction)
+					    ? UDMA_MODE_PER_ALT_SCATTER_GATHER
+					    : UDMA_MODE_MEM_ALT_SCATTER_GATHER;
+		/* Set the arbitration size */
+		cfg->rPower = ARBSIZE_1;
+		if (siwx917_addr_adjustment(block_addr->source_addr_adj) < 0 ||
+		    siwx917_addr_adjustment(block_addr->dest_addr_adj) < 0) {
+			return -EINVAL;
+		}
+		/* Set source and destination address increments */
+		cfg->srcInc = siwx917_addr_adjustment(block_addr->source_addr_adj)
+				      ? UDMA_SRC_INC_NONE
+				      : siwx917_data_width(config->source_data_size);
+		cfg->dstInc = siwx917_addr_adjustment(block_addr->dest_addr_adj)
+				      ? UDMA_DST_INC_NONE
+				      : siwx917_data_width(config->dest_data_size);
+		/* Move to the next block */
+		block_addr = block_addr->next_block;
+	}
+	if (block_addr != NULL) {
+		/* next_block address for last block must be null */
+		return -EINVAL;
+	}
+	/* Set the transfer type for the last descriptor */
+	descs[config->block_count - 1].vsUDMAChaConfigData1.transferType =
+		siwx917_transfer_direction(config->channel_direction) ? UDMA_MODE_BASIC
+									 : UDMA_MODE_AUTO;
+	return 0;
+}
+
+/* Configure DMA for scatter-gather transfer */
+static int siwx917_sg_config(const struct device *dev, RSI_UDMA_HANDLE_T udma_handle,
+			     uint32_t channel, struct dma_config *config)
+{
+	uint8_t transfer_type = UDMA_MODE_MEM_SCATTER_GATHER;
+	const struct dma_siwx917_config *cfg = dev->config;
+	struct dma_siwx917_data *data = dev->data;
+	RSI_UDMA_DESC_T *sg_desc_base_addr = NULL;
+
+	if (siwx917_transfer_direction(config->channel_direction) == 1) {
+		transfer_type = UDMA_MODE_PER_SCATTER_GATHER;
+	} else if (siwx917_transfer_direction(config->channel_direction) < 0) {
+		return -EINVAL;
+	}
+	if (siwx917_data_width(config->source_data_size) < 0 ||
+	    siwx917_data_width(config->dest_data_size) < 0) {
+		return -EINVAL;
+	}
+	if (config->block_count > CONFIG_DMA_SILABS_SIWX917_SG_BUFFER_COUNT) {
+		return -EINVAL;
+	}
+	/* Request start index for scatter-gather descriptor table */
+	if (sys_mem_blocks_alloc_contiguous(data->dma_desc_pool, config->block_count,
+					    (void **)&sg_desc_base_addr)) {
+		return -EINVAL;
+	}
+	if (siwx917_sg_fill_desc(sg_desc_base_addr, config)) {
+		return -EINVAL;
+	}
+	/* This channel information is used to distinguish scatter-gather transfers and
+	 * free the allocated descriptors in sg_transfer_desc_block
+	 */
+	data->chan_info[channel].SrcAddr = 0;
+	data->chan_info[channel].DestAddr = 0;
+	data->chan_info[channel].Cnt = config->block_count;
+	data->chan_info[channel].Size = (uint32_t)sg_desc_base_addr;
+	RSI_UDMA_InterruptClear(udma_handle, channel);
+	RSI_UDMA_ErrorStatusClear(udma_handle);
+	if (cfg->reg == UDMA0) {
```

Comment:
In addition to my other suggestion, I think it'd be logical to have an empty line between these two lines.